### PR TITLE
adding templates 

### PR DESCRIPTION
**Pull request by fengyilu**
I created 3 different Templates, by adding the extension from GOJS called Figures.JS.
we have to think about, if we should put the template files into a diffrent Js file or not. 
I tried to do this but the problem is that we cannot Export and import variables on the Client side with javascript.

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
+    $(go.Node, "Vertical",  // "vertical" determine the order of the go.objects 
+        $(go.Shape, "Database", { height: 70, width: 70, fill: "blue" }, //per default color and size is setted, can be customised through e.g. "go.Binding()"
+            new go.Binding("fill", "color")),
+        $(go.TextBlock, { font: "bold 12pt sans-serif" },
+            new go.Binding("text", "key")),
+        $(go.TextBlock, "Description:"),
+        $(go.TextBlock, new go.Binding("text", "desc")),
+
+    );
+
```
Comment (antonroesler) [inline public/js/index.js:21]: Please use 
```javascript
/**
* Comments to explain what a template is used for and how to use it 
*/
```

**Review by antonroesler — CHANGES_REQUESTED** [review]
1. Please remove Figures.js, we cannot publish gojs files on github
2. More comments are needed - the code must be easily understandable by anyone
3. Please add the header from misc/header.txt

**Comment by antonroesler** [comment]
And also there should be a way to separate the templates into another file.

File: public/js/templates.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+* University:		 Frankfurt University of Applied Sciences
+* Study program:	 Engineering Business Information Systems
+* Module:		   Advanced Programming 2021
+* Professor:		   Prof. Dr. Jung, Prof. Dr. Bremm
+* Date:			     21.04.2021
+*
+*/
+
+/**
+* A short description what this file/class is all about.
```
Comment (antonroesler) [inline public/js/templates.js:13]: Remove this sentence and put the description here. Also make sure the lines are not longer than 80 characters.

File: public/js/templates.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* @author Feng Yi Lu
+*this file contains all templates that will be used for different Nodes
+*- templates are used for various Nodes, its a configuration for a certain type of Node e.g. Database
+*- usage: in this file we create a variable that contains a preset of a Node-style
+*-        these variables can be used in the index.js file. 
+*-        "templmap" is a Map that contains all of our templates that we create, with the function add("category-name", template-variable) we can add more templates to it. 
+*-        In Index.js we declare a TemplateMap to our diagram "diagram.nodeTemplateMap = templmap;" with that our nodes are able to use these templates (the attribute "category" inside of the Node Object declares which template it uses)
+*        
+*/
+
```
Comment (antonroesler) [inline public/js/templates.js:23]: We still need a comment for every template to ecplain what it it used for:
```javascript
/**
* Template ...
* used for...
*/
```

File: public/js/templates.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+var databaseTemplate =
+    $(go.Node, "Vertical",  // "vertical" determine the order of the go.objects 
+        $(go.Shape, "Database", { height: 70, width: 70, fill: "blue" }, //per default color and size is setted, can be customised through e.g. "go.Binding()"
+            new go.Binding("fill", "color")),
+        $(go.TextBlock, { font: "bold 12pt sans-serif" },
+            new go.Binding("text", "key")),
+        $(go.TextBlock, "Description:"),
+        $(go.TextBlock, new go.Binding("text", "desc")),
+    );
+
```
Comment (antonroesler) [inline public/js/templates.js:33]:
```javascript
/**
* Template ...
* used for...
*/
```

File: public/js/templates.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    $(go.Node, "Vertical",
+        $(go.Shape, "Component", { height: 70, width: 70, fill: "red" },
+            new go.Binding("fill", "color")),
+
+        $(go.TextBlock, { font: "bold 12pt sans-serif" },
+            new go.Binding("text", "key")),
+        $(go.TextBlock, "Description:"),
+        $(go.TextBlock, new go.Binding("text", "desc")),
+    );
+
```
Comment (antonroesler) [inline public/js/templates.js:44]:
```javascript
/**
* Template ...
* used for...
*/
```

**Review by antonroesler — CHANGES_REQUESTED** [review]

**Comment by antonroesler** [comment]
I have corrected the comments. The comment should be above the function and only describe what our function does, not explain how e.g. a GoJS function works.

**Review by antonroesler — APPROVED** [review]
its ok for now, we may need some more refactoring later

**Comment by antonroesler** [comment]
Please someone else participate in the approval process!! We always need two people to approve something. Of course, do not simply approve. Please look for errors or otherwise comment and make a request change instead of approve. Only approve if everything is ok. ✌🏻

**Review by leo-hsk — APPROVED** [review]

**Pull request merged**